File: undo_copy.py

import os, shutil

source = "/tmp/test/source" # sys.argv[1]
target = "/tmp/test/target" # sys.argv[2]

files = os.listdir(source) # returns a list of top level directory contents

for file in files:
    to_del = os.path.join(target, file) # join target with source file/directory
    if os.path.isdir(to_del):
        print(f"Removing Directory: {to_del}")
        try:
            shutil.rmtree(to_del) # remove directories and sub-directories
        except FileNotFoundError:
            pass
    else:
        print(f"Removing File: {to_del}")
        try:
            os.remove(to_del) # remove files
        except FileNotFoundError:
            pass

print("Done")

File: undo_copy.go
package main

import (
	"fmt"
	"os"
	path "path/filepath"
)

func check(err error) {
	if err != nil {
		panic(err)
	}
}

func main() {
	source := "/tmp/test/source" // os.Argv[1]
	target := "/tmp/test/target" // os.Argv[1]
	files, err := os.ReadDir(source)
	check(err)

	for _, file := range files {
		to_del := path.Join(target, file.Name())
		if file.IsDir() {
			fmt.Printf("Removing Directory: %v\n", to_del)
			err := os.RemoveAll(to_del)
			check(err)
		} else {
			fmt.Printf("Removing File: %v\n", to_del)
			err := os.Remove(to_del)
			check(err)
		}

	}
}